Replace `execution_timeout` with `timeout` in sensor

rajaths010494 opened the pull request:
Use `timeout` instead of `execution_timeout` in Async Sensors for deferrable timeout arg

Sensors have a timeout parameter. This only matters for sensors in reschedule mode. timeout controls the maximum time allowed for the sensor to succeed. If the timeout is breached, AirflowSensorTimeout will be raised and the sensor fails immediately without retrying. So when we defer for sensor we need to send timeout rather than sending execution._timeout.

Inherited BaseSensorOperator because the timeout is a parameter in that class, so we inherited that and use it when we defer.

Closes #513

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,11 @@ Goals for the project
+Limitations
+-----------
+
+- Async sensors take param ``mode`` but it doesn't have any usage in deferrables.
```
did not understand this, can you elaborate please?

rajaths010494 replied:
yes, sure I will add more details to it.

rajaths010494 replied:
i have changed it .

phanikumv commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
+if sync version isn't available then from airflow ``BaseOperator``.
+
+To create a async sensors we need to inherit from sync sensors or
+if sync version isn't available then from airflow ``BaseSenseOperator``.
```
```suggestion
if sync version isn't available then from airflow ``BaseSensorOperator``.
```

phanikumv commented on astronomer/providers/google/cloud/sensors/gcs.py:
```diff
@@ -39,32 +40,18 @@ class GCSObjectExistenceSensorAsync(BaseOperator):
         account from the list granting this role to the originating account (templated).
     """
 
-    template_fields = ("bucket", "object", "impersonation_chain")
```
why are we removing `template_fields`

rajaths010494 replied:
Initially, it was inherited from `BaseOperator` now it's from `GCSObjectExistenceSensor` which contains this information already. So I have removed the `template_fields`.

phanikumv commented on astronomer/providers/google/cloud/sensors/gcs.py:
```diff
@@ -39,32 +40,18 @@ class GCSObjectExistenceSensorAsync(BaseOperator):
-
     def __init__(
         self,
-        *,
-        bucket: str,
-        object: str,  # noqa: A002
```
why are we removing these? Have you tested all these sensors and ensured that the functionality remains as earlier?

rajaths010494 replied:
Yes, I have run all the sensors dags and this variable is present in `GCSObjectExistenceSensor`  which is being inherited instead of `BaseOperator`.

bharanidharan14 commented:
> Use `timeout` instead of `execution_timeout` in Async Sensors for deferrable timeout arg
> 
> Closes #513

Can you explain why `timeout` is used  instead of `execution_timeout` and why the BaseSensorOperator is inhertited

pankajkoti commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
+if sync version isn't available then from airflow ``BaseOperator``.
+
+To create a async sensors we need to inherit from sync sensors or
+if sync version isn't available then from airflow ``BaseSenseOperator``.
```
Did we have a discussion on whether Async sensors should inherit from BaseOperator or BaseSensorOperator ? 
I read somewhere @dstandish mentioning that async sensors should inherit from BaseOperator

rajaths010494 replied:
Yes i had a discussion with @kaxil  during the meetup, he suggested moving all sensors under BaseSenseOperator and also creating a ticket for this particular issue which you mentioned in the backlog and then deciding whether it's BaseOpertaor vs BaseSensorOperator?

rajaths010494 replied:
Created a ticket for the discussion on the same[ here](https://app.zenhub.com/workspaces/airflow-providers-team-61a89ee7fbf048001dc8378d/issues/astronomer/astronomer-providers/558).

bharanidharan14 reviewed (commented):
Can you test once after this changes

bharanidharan14 commented on astronomer/providers/amazon/aws/sensors/s3.py:
```diff
@@ -200,39 +201,16 @@ class S3KeysUnchangedSensorAsync(BaseOperator):
         **kwargs: Any,
     ) -> None:
         super().__init__(**kwargs)
-        self.bucket_name = bucket_name
```
why are this params are been removed ?

rajaths010494 replied:
We were using BaseOperator before so we had added this now it's from `S3KeysUnchangedSensor` so due to inheritance these variables are already present.

rajaths010494 commented:
> > Use `timeout` instead of `execution_timeout` in Async Sensors for deferrable timeout arg
> > Closes #513
> 
> Can you explain why `timeout` is used instead of `execution_timeout` and why the BaseSensorOperator is inherited

Sensors have a timeout parameter. This only matters for sensors in reschedule mode. timeout controls the maximum time allowed for the sensor to succeed. If the timeout is breached, AirflowSensorTimeout will be raised and the sensor fails immediately without retrying. So when we defer for sensor we need to send timeout rather than sending execution._timeout.

Inherited BaseSensorOperator because the timeout is a parameter in that class, so we inherited that and use it when we defer.

rajaths010494 commented:
> Can you test once after this changes

I tested all the sensor dags after the changes. I can retest if needed.

pankajkoti commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
 For example, we won’t create an async Operator for a ``BigQueryCreateEmptyTableOperator`` but will create one
 for ``BigQueryInsertJobOperator`` that actually runs queries and can take hours in the worst case for task completion.
 
+To create a async operators we need to inherit from airflow sync operators or
+if sync version isn't available then from airflow ``BaseOperator``.
```
```suggestion
To create async operators, we need to inherit from airflow sync operators.
If sync version isn't available, then inherit from airflow ``BaseOperator``.
```

pankajkoti commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
+To create a async operators we need to inherit from airflow sync operators or
+if sync version isn't available then from airflow ``BaseOperator``.
+
+To create a async sensors we need to inherit from sync sensors or
+if sync version isn't available then from airflow ``BaseSensorOperator``.
```
```suggestion
To create async sensors, we need to inherit from sync sensors.
If sync version isn't available, then inherit from airflow ``BaseSensorOperator``.
```

bharanidharan14 commented:
> Sensors have a timeout parameter. This only matters for sensors in reschedule mode. timeout controls the maximum time allowed for the sensor to succeed. If the timeout is breached, AirflowSensorTimeout will be raised and the sensor fails immediately without retrying. So when we defer for sensor we need to send timeout rather than sending execution._timeout.
> 
> Inherited BaseSensorOperator because the timeout is a parameter in that class, so we inherited that and use it when we defer.

Copying this in the description

bharanidharan14 commented:
LGTM @pankajkoti @phanikumv @pankajastro what do you guys think?

bharanidharan14 commented:
@phanikumv can you please take a look at the PR review changes ?

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,13 @@ Goals for the project
+Limitations
+-----------
+
+- In Airflow sensors have a param ``mode`` which has two types poke and reschdule,
```
```suggestion
- In Airflow, sensors have a param ``mode``, which can be  ``poke`` and ``reschedule``
```

rajaths010494 replied:
Fixed

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,13 @@ Goals for the project
+-----------
+
+- In Airflow sensors have a param ``mode`` which has two types poke and reschdule,
+  which uses the worker slot to poke or at intervals respectively. Whereas in async sensors this param has
```
This sentence needs to be grammatically corrected as mentioned above, and avoid writing big sentences please

rajaths010494 replied:
Changed it.

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,12 @@ Goals for the project
+-----------
+
+- In Airflow sensors have a param ``mode`` which can be ``poke`` and ``reschedule``,
+  Whereas in async sensors this param has no usage since tasks gets deferred to Trigger.
```
```suggestion
  whereas in async sensors, this param has no usage since tasks gets deferred to Trigger.
```

bharanidharan14 commented:
@rajaths010494 I hope you have addressed the Review comments on this PR, Need your confirmation?

rajaths010494 commented:
> @rajaths010494 I hope you have addressed the Review comments on this PR, Need your confirmation?

Yes, I have addressed the review comments and also rebased the branch.

phanikumv commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
 For example, we won’t create an async Operator for a ``BigQueryCreateEmptyTableOperator`` but will create one
 for ``BigQueryInsertJobOperator`` that actually runs queries and can take hours in the worst case for task completion.
 
+To create async operators, we need to inherit from airflow sync operators.
```
```suggestion
To create async operators, we need to inherit from the corresponding airflow sync operators.
```

rajaths010494 replied:
fixed

phanikumv commented on README.rst:
```diff
@@ -136,6 +136,12 @@ We will only create Async operators for the "sync-version" of operators that do
+To create async operators, we need to inherit from airflow sync operators.
+If sync version isn't available, then inherit from airflow ``BaseOperator``.
+
+To create async sensors, we need to inherit from sync sensors.
```
```suggestion
To create async sensors, we need to inherit from the corresponding sync sensors.
```

rajaths010494 replied:
fixed

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,12 @@ Goals for the project
+Limitations
+-----------
+
+- In Airflow sensors have a param ``mode`` which can be ``poke`` and ``reschedule``,
```
```suggestion
- In Airflow sensors have a param ``mode`` which can be ``poke`` and ``reschedule``.
```

rajaths010494 replied:
fixed

phanikumv commented on README.rst:
```diff
@@ -166,6 +172,12 @@ Goals for the project
+-----------
+
+- In Airflow sensors have a param ``mode`` which can be ``poke`` and ``reschedule``,
+  whereas in async sensors, this param has no usage since tasks gets deferred to Trigger.
```
```suggestion
  In async sensors, this param has no usage since tasks gets deferred to Triggerer.
```

rajaths010494 replied:
fixed

bharanidharan14 commented:
Waiting for the release to close this
